src/pyproject_fmt/formatter/pep508.py

```diff
@@ -39,18 +39,25 @@ def _best_effort_string_repr(req: str) -> String:
         return toml_string(req)
 
 
-def normalize_pep508_array(requires_array: Array | None, indent: int) -> None:
+def normalize_pep508_array(requires_array: Array | None, indent: int, *, preserve_dependency_versions: bool) -> None:
     """
     Normalize a TOML array via PEP-508.
 
     :param requires_array: the input array
     :param indent: indentation level
+    :param preserve_dependency_versions: whether to preserve, and therefore not normalize, requirements
     """
     if requires_array is None:
         return
     # first normalize values
     for at in range(len(requires_array)):
-        normalized = _best_effort_string_repr(normalize_req(str(requires_array[at])))
+        initial_requirement_string = str(requires_array[at])
+        if preserve_dependency_versions:
+            final_requirement_string = initial_requirement_string
+        else:
+            final_requirement_string = normalize_req(initial_requirement_string)
```

Comment:
can we make this one line?

Reply:
Kind of - but `ruff` moved it to be in parentheses.